packages/features/bookings/lib/handleNewBooking/loadAndValidateUsers.ts

```diff
@@ -96,6 +97,12 @@ export async function loadAndValidateUsers({
   }
 
   if (!users) throw new HttpError({ statusCode: 404, message: "eventTypeUser.notFound" });
+
+  // Determine if users are locked
+  const containsBlockedUser = await checkIfUsersAreBlocked(users);
```

Comment:
If at least one user is blocked then fail the whole booking. Prevents abusers from bypassing dynamic bookings